### ISSUE TITLE
dune-configurator 2.5.1 fails to build

### ISSUE THREAD

**Issue by robz** [ocaml/dune#3625]
## Expected Behavior
dune-configurator 2.5.1 should not fail to build

## Actual Behavior
I receive a build error when attempting to build with dune-configurator 2.5.1: https://raw.githubusercontent.com/robz/repro-dune-configurator-build-failure/master/error.txt

## Reproduction

Clone https://github.com/robz/repro-dune-configurator-build-failure/ 
Then run `esy`

## Specifications

- Version of `dune` (output of `dune --version`): From esy.lock folder, it seems to be dune.2.6.1: https://github.com/robz/repro-dune-configurator-build-failure/tree/master/esy.lock/opam
- Version of `ocaml` (output of `ocamlc --version`): not sure, but the esy.lock file has "ocaml-secondary-compiler.4.08.1-1": https://github.com/robz/repro-dune-configurator-build-failure/tree/master/esy.lock/opam
- Operating system (distribution and version): macOS Catalina 10.15.5

**Comment by rgrinberg** [comment]
The latest version of configurator fixes this issue. Re-open if it does not.

**Comment by robz** [comment]
Upgrading to 2.6.0+ fixes it, thanks! 

But the build failure also occurs without explicitly setting the version of dune-configurator to 2.5.1. I updated the repo with a minimal repro: https://github.com/robz/repro-dune-configurator-build-failure/

```
  "dependencies": {
    "@opam/lwt": "*",
    "ocaml": "~4.8.0"
  }
```

It seems that esy automatically resolves dune-configurator to 2.5.1, so adding an override to package.json resolutions is required. Is there something that could be done to avoid automatically resolving to dune-configurator v2.5.1 in this case?

**Comment by rgrinberg** [comment]
You could a conflict I suppose? The conflict should be between dune-configuration 2.5.x and dune-private-libs >= 2.6.x.

Sorry, I’m not familiar enough with esy to suggest something with confidence.
